tests/framework/installer/ceph_installer.go

```diff
@@ -235,6 +235,122 @@ func (h *CephInstaller) CreateK8sRookClusterWithHostPathAndDevices(namespace, sy
+	}
+	logger.Infof("Creating external cluster: namespace=%q, firstClusterNamespace=%q", namespace, firstClusterNamespace)
+
+	logger.Infof("Creating namespace %s", namespace)
+	ns := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
+	_, err = h.k8shelper.Clientset.CoreV1().Namespaces().Create(ns)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("failed to create namespace %q. %v", namespace, err)
+	}
+
+	// Skip this step since the helm chart already includes the roles and bindings
+	if !h.useHelm {
+		logger.Infof("Creating external cluster roles")
+		roles := h.Manifests.GetClusterExternalRoles(namespace, firstClusterNamespace)
+		if _, err := h.k8shelper.KubectlWithStdin(roles, createFromStdinArgs...); err != nil {
+			return fmt.Errorf("failed to create cluster roles. %v", err)
+		}
+	}
+
+	// Inject connection information from the first cluster
+	logger.Info("Injecting cluster connection information")
+	err = h.InjectRookExternalClusterInfo(namespace, firstClusterNamespace)
+	if err != nil {
+		return fmt.Errorf("failed to inject cluster information into the external cluster. %v", err)
+	}
+
+	// Start the external cluster
+	logger.Infof("Starting Rook External Cluster with yaml")
+	settings := &ClusterExternalSettings{namespace, dataDirHostPath}
+	rookCluster := h.Manifests.GetRookExternalCluster(settings)
+	if _, err := h.k8shelper.KubectlWithStdin(rookCluster, createFromStdinArgs...); err != nil {
+		return fmt.Errorf("failed to create rook external cluster. %v ", err)
+	}
+
+	logger.Infof("Rook external cluster started")
+	return err
+}
+
+// InjectRookExternalClusterInfo inject connection information for an external cluster
+func (h *CephInstaller) InjectRookExternalClusterInfo(namespace, firstClusterNamespace string) error {
+	args := []string{"apply", "--namespace", namespace, "-f"}
+
+	// get cm
+	cm, err := h.GetRookExternalClusterMonConfigMap(firstClusterNamespace)
+	if err != nil {
+		return fmt.Errorf("failed to get secret. %v", err)
+	}
+
+	// get secret
+	secret, err := h.GetRookExternalClusterMonSecret(firstClusterNamespace)
+	if err != nil {
+		return fmt.Errorf("failed to get configmap. %v", err)
+	}
+
+	// create cm
+	_, err = h.k8shelper.KubectlWithStdin(cm, args...)
+	if err != nil {
+		return fmt.Errorf("failed to create cm. %v", err)
+
+	}
+
+	// create secret
+	_, err = h.k8shelper.KubectlWithStdin(secret, args...)
+	if err != nil {
+		return fmt.Errorf("failed to create secret. %v", err)
+	}
+
+	return nil
+}
+
+// GetRookExternalClusterMonConfigMap gets the monitor kubernetes configmap of the external cluster
+func (h *CephInstaller) GetRookExternalClusterMonConfigMap(namespace string) (string, error) {
+	configMapName := "rook-ceph-mon-endpoints"
+	args := []string{"--namespace", namespace, "get", "configmap", configMapName, "--export", "-o", "yaml"}
+
+	cm, err := h.k8shelper.Kubectl(args...)
+	if err != nil {
+		debugArgs := []string{"--namespace", namespace, "get", "configmap"}
```

Comment:
Is all of this debug still needed? After the test is initially working I would think this extra debug code wouldn't be needed.